### PR TITLE
[Bug] remote mode job remapping status bug fixed

### LINKED ISSUE #2818: [Bug] flink集群重启，streampark对应任务的状态lost
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### Java Version

1.8

### Scala Version

2.11.x

### StreamPark Version

2.0.0

### Flink Version

1.15.2

### deploy mode

remote

### What happened

环境：flink集群的standalone模式，两个jobmanager的HA模式

问题场景：
1、streampark提交运行A任务
2、flink集群stop，streampark的A任务状态显示lost
3、flink集群start，A任务自动重启，但是streampark的A任务状态依然显示lost
4、使用remapping功能，对应当前jobid，任务一直处于mapping状态

### Error Exception

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!(您是否要贡献这个PR?)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wolfboys**
remote mode job remapping status bug fixed

## What changes were proposed in this pull request

Issue Number: close #2818

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkRESTAPIWatcher.java [hunk 1]
```diff
@@ -700,30 +683,31 @@ private JobsOverview httpJobsOverview(Application application, FlinkCluster flin
         reqURL = String.format(format, application.getJobManagerUrl());
       }
       return yarnRestRequest(reqURL, JobsOverview.class);
-    } else if (ExecutionMode.REMOTE.equals(execMode)
-        || ExecutionMode.YARN_SESSION.equals(execMode)) {
+    } else if (ExecutionMode.REMOTE.equals(execMode)) {
```
Comment (zhoulii) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkRESTAPIWatcher.java]:
```suggestion
    } else if (ExecutionMode.isRemoteMode(execMode)) {
```

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkRESTAPIWatcher.java [hunk 2]
```diff
@@ -733,12 +717,15 @@ private CheckPoints httpCheckpoints(Application application, FlinkCluster flinkC
         reqURL = String.format(format, application.getJobManagerUrl(), application.getJobId());
       }
       return yarnRestRequest(reqURL, CheckPoints.class);
-    } else if (ExecutionMode.REMOTE.equals(execMode)
-        || ExecutionMode.YARN_SESSION.equals(execMode)) {
+    } else if (ExecutionMode.REMOTE.equals(execMode)) {
```
Comment (zhoulii) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkRESTAPIWatcher.java]:
```suggestion
    } else if (ExecutionMode.isRemoteMode(execMode)) {
```

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkRESTAPIWatcher.java [hunk 3]
```diff
@@ -765,4 +752,19 @@ private <T> T httpRestRequest(String url, Class<T> clazz) throws IOException {
   public boolean isWatchingApp(Long id) {
     return WATCHING_APPS.containsKey(id);
   }
+
+  private <T> T httpClusterOrElseTry(Long clusterId, Callback<FlinkCluster, T> function)
```
Comment (zhoulii) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkRESTAPIWatcher.java]: nice design.

**Review by zhoulii — COMMENTED** [review]
Overall, It looks good me, just left a few minor comments.

**Comment by wolfboys** [comment]
cc @zhoulii updated , PTAL again, thx

**Review by zhoulii — APPROVED** [review]
LGTM

**Pull request merged**